### PR TITLE
APPS-2671 Custom Chip

### PR DESCRIPTION
**Pull request by bergieboy**
Screen Shot 2021-03-18 at 6 38 56 PM

### CODE REVIEW

**Review by solomontang — APPROVED** [review]

**Review by giuliacampana — APPROVED** [review]

File: src/chip/src/components/Chip.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    deleteIcon,
+    disabled,
+    icon,
+    label,
+    onDelete,
+    size,
+    variant,
+  };
+
+  if (color === 'default' || color === 'primary' || color === 'secondary') props.color = color;
```
Comment (UnsungHero97) [inline src/chip/src/components/Chip.js:72]: What does this line do? It doesn't seem like this has any effect when running it locally.

File: src/chip/src/components/colors.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    clickable: YELLOW.Y100
+  },
+  primary: {
+    background: PURPLE.P300,
+    color: 'white'
+  },
+  secondary: {
+    background: PURPLE.P00,
+    color: PURPLE.P400
+  },
```
Comment (UnsungHero97) [inline src/chip/src/components/colors.js:73]: `purple` and `secondary` look identical. Why does `secondary` need to exist?

Screen Shot 2021-03-22 at 12 51 15 PM

File: src/chip/src/components/types.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  | 'blue'
+  | 'green'
+  | 'yellow'
+  | 'orange'
+  | 'red'
+  | 'pink'
+  | 'violet';
+
+type ChipSize =
+  'medium'
```
Comment (UnsungHero97) [inline src/chip/src/components/types.js:19]: Missing `|`, should be `| medium`.

File: src/chip/src/components/types.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  color ?:ChipColor;
+  component ?:string;
+  deleteIcon ?:Node;
+  disabled ?:boolean;
+  icon ?:Node;
+  label ?:string;
+  onDelete ?:Function;
+  size ?:ChipSize;
+  variant ?:ChipVariant;
+  forwardRef ?:any;
```
Comment (UnsungHero97) [inline src/chip/src/components/types.js:39]: Sort.

File: src/chip/src/components/Chip.test.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    });
+
+  });
+
+  test('default', () => {
+    const wrapper = mount(<Chip />);
+    const chip = wrapper.find('div');
+    expect(chip).toHaveLength(1);
+    expect(window.getComputedStyle(chip.getDOMNode())).toHaveProperty('color', hexToRgb(NEUTRAL.N600));
+    expect(window.getComputedStyle(chip.getDOMNode())).toHaveProperty('background', hexToRgb(NEUTRAL.N00));
```
Comment (UnsungHero97) [inline src/chip/src/components/Chip.test.js:27]: I haven't seen `window.getComputedStyle` and `getDOMNode()` before, interesting. How did you come across this? 

Also, just curious, does the following work?

```
const chip = wrapper.find(Chip);
expect(chip).toHaveStyleRule('color', NEUTRAL.N600);
expect(chip).toHaveStyleRule('background', NEUTRAL.N00);
```

File: src/chip/src/components/Chip.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    },
+    clickable: {
+      cursor: 'pointer',
+      ...actionStates(colors.clickable)
+    },
+    deletable: actionStates(colors.background),
+    deleteIcon: {
+      color: colors.color,
+      margin: '0 12px 0 -6px',
+    }
```
Comment (UnsungHero97) [inline src/chip/src/components/Chip.js:35]: I think there's an issue with the `deleteIcon` style. It seems MUI sets a `height` which makes the icon look bigger than it should be.

Screen Shot 2021-03-22 at 12 58 31 PM

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

File: src/chip/src/components/Chip.js (hunk 6 above)
Comment (UnsungHero97) [reply src/chip/src/components/Chip.js:35]: Update: removing `fixedWidth` seems to fix this issue.

File: src/chip/src/components/Chip.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+const actionStates = (backgroundColor) => ({
+  '&:hover, &:focus': {
+    backgroundColor
+  },
+  '&:active': {
+    backgroundColor
+  },
+});
+
+const useStyles = (color) => {
```
Comment (UnsungHero97) [inline src/chip/src/components/Chip.js:21]: The hover color for the delete icon is gray for all colors except `primary`, which doesn't look good, and it's a dark purple for `primary`, which also does not look good. Please update.

Screen Shot 2021-03-22 at 1 16 48 PM
Screen Shot 2021-03-22 at 1 14 04 PM

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

File: src/chip/src/components/colors.js (hunk 2 above)
Comment (bergieboy) [reply src/chip/src/components/colors.js:73]: secondary is default mui based on our theme. I can remove either.

File: src/chip/src/components/Chip.test.js (hunk 5 above)
Comment (bergieboy) [reply src/chip/src/components/Chip.test.js:27]: `const chip = wrapper.find(Chip);` works, but the following lines doo not. I found getComputedStyle when researching how to test css attributes of MUI components.

File: src/chip/src/components/colors.js (hunk 2 above)
Comment (bergieboy) [reply src/chip/src/components/colors.js:73]: I'm removing color purple

File: src/chip/src/components/Chip.js (hunk 1 above)
Comment (bergieboy) [reply src/chip/src/components/Chip.js:72]: This prevents console errors from passing invalid color prop to mui chip.

**Review by UnsungHero97 — APPROVED** [review]

**Review by giuliacampana — APPROVED** [review]

**Review by solomontang — APPROVED** [review]

**Pull request merged**